pkg/accumulator/accumulator.go

```diff
@@ -0,0 +1,221 @@
+	evData "sniffer/pkg/ebpfev/v1"
+
+	logger "github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+)
+
+const (
+	CACHE_ACCUMULATOR_SIZE = 10
+)
+
+type containersAccumulator struct {
+	accumulatorDataPerContainer map[string]chan evData.EventData
+	registerMutex               sync.Mutex
+}
+
+type CacheAccumulator struct {
+	accumulatorData                 []map[string][]evData.EventData
+	syncReaderWriterAccumulatorData sync.Mutex
+	firstMapKeysOfAccumulatorData   []string
+	cacheAccumulatorSize            int
+	mainDataChannel                 chan *evData.EventData
+	containersData                  containersAccumulator
+	ebpfEngine                      ebpfeng.EbpfEngineClient
+}
+
+type ContainerAccumulator struct {
+	dataChannel chan evData.EventData
+	containerID string
+}
+
+var myContainerID string
+var cacheAccumulator *CacheAccumulator
+
+func CreateAccumulator() *CacheAccumulator {
+	cacheAccumulator = &CacheAccumulator{
+		cacheAccumulatorSize:          CACHE_ACCUMULATOR_SIZE,
+		accumulatorData:               make([]map[string][]evData.EventData, CACHE_ACCUMULATOR_SIZE),
+		firstMapKeysOfAccumulatorData: make([]string, CACHE_ACCUMULATOR_SIZE),
+		mainDataChannel:               make(chan *evData.EventData),
+		containersData: containersAccumulator{
+			accumulatorDataPerContainer: make(map[string]chan evData.EventData),
+		},
+	}
+
+	return cacheAccumulator
+}
+
+func CreateContainerAccumulator(containerID string, dataChannel chan evData.EventData) *ContainerAccumulator {
+	return &ContainerAccumulator{
+		dataChannel: dataChannel,
+		containerID: containerID,
+	}
+}
+
+func (acc *CacheAccumulator) createNewMap(event *evData.EventData, index int) {
+	slice := make([]evData.EventData, 0)
+	m := make(map[string][]evData.EventData)
+	m[event.GetEventContainerID()] = slice
+	acc.accumulatorData[index] = m
+	acc.firstMapKeysOfAccumulatorData[index] = event.GetEventContainerID()
+}
+
+func (acc *CacheAccumulator) findIndexByTimestampWhenAccumulatorDataIsFull(event *evData.EventData) int {
+	index := 0
+	minTimestamp := acc.accumulatorData[0][acc.firstMapKeysOfAccumulatorData[0]][0].GetEventTimestamp()
+	for i := range acc.accumulatorData {
+		if i == 0 {
+			continue
+		}
+		if acc.accumulatorData[i][acc.firstMapKeysOfAccumulatorData[i]][0].GetEventTimestamp().Before(minTimestamp) {
+			minTimestamp = acc.accumulatorData[i][acc.firstMapKeysOfAccumulatorData[i]][0].GetEventTimestamp()
+			index = i
+		}
+	}
+	acc.createNewMap(event, index)
+	return index
+}
+
+func (acc *CacheAccumulator) findIndexByTimestamp(event *evData.EventData) int {
+	for i := range acc.accumulatorData {
+		if len(acc.accumulatorData[i]) == 0 {
+			acc.createNewMap(event, i)
+			return i
+		}
+		firstKey := acc.firstMapKeysOfAccumulatorData[i]
+		if event.GetEventTimestamp().Sub((acc.accumulatorData[i])[firstKey][0].GetEventTimestamp()) < time.Second {
+			return i
+		}
+	}
+	index := acc.findIndexByTimestampWhenAccumulatorDataIsFull(event)
```

Comment:
by the way `findIndexByTimestampWhenAccumulatorDataIsFull` there is no possibility to get `index = -1` ...